Sources/Sockets/TCP/TCPReadableSocket.swift

```diff
@@ -3,27 +3,25 @@ import libc
 public protocol TCPReadableSocket: TCPSocket, ReadableStream {}
 
 extension TCPReadableSocket {
-    public func read(max: Int) throws -> Bytes{
-        let buffer = Buffer(capacity: max)
-
+    public func read(max: Int, into buffer: inout Bytes) throws -> Int {
         let receivedBytes = libc.recv(
```

Comment:
Since we're making a call to `recv` but it's not possible for outside users to pass in any flags, I would argue that it would be more consistent to use [`read`](http://man7.org/linux/man-pages/man2/read.2.html) here instead. If you check the manpage for [recv](http://man7.org/linux/man-pages/man2/recv.2.html) you can see the following:

<img width="681" alt="screen shot 2017-04-30 at 10 20 31 pm" src="https://cloud.githubusercontent.com/assets/1977704/25567671/4be36ef8-2df3-11e7-89d9-8aa319799375.png">